charts/prometheus/templates/alertmanager/deploy.yaml

```diff
@@ -47,8 +47,12 @@ spec:
         - name: {{ template "prometheus.name" . }}-{{ .Values.alertmanager.name }}
           image: "{{ .Values.alertmanager.image.repository }}:{{ .Values.alertmanager.image.tag }}"
           imagePullPolicy: "{{ .Values.alertmanager.image.pullPolicy }}"
+          {{- if .Values.alertmanager.containerSecurityContextEnabled }}
```

Comment:
the containerSecurityContextEnabled variable is superfluous.
just use it like this:
```
  {{- with .Values.alertmanager.containerSecurityContext }}
  securityContext:
    {{- toYaml . | nindent 12 }}
  {{- end }}
```

Reply:
@monotek Please check this [comment](https://github.com/prometheus-community/helm-charts/pull/2595#issuecomment-1290917129)

Reply:
my proposed change does exactly that, without introducing a new var.